runtime/queries/metricsview.go

```diff
@@ -292,6 +292,327 @@ func buildFilterClauseForCondition(mv *runtimev1.MetricsViewSpec, cond *runtimev
+	}
+
+	return "", emptyArg, nil
+}
+
+func buildFromConditionExpression(cond *runtimev1.Condition, allowedIdentifiers map[string]identifier, dialect drivers.Dialect) (string, []any, error) {
+	switch cond.Op {
+	case runtimev1.Operation_OPERATION_LIKE, runtimev1.Operation_OPERATION_NLIKE:
+		return buildFromLikeExpression(cond, allowedIdentifiers, dialect)
+
+	case runtimev1.Operation_OPERATION_IN, runtimev1.Operation_OPERATION_NIN:
+		return buildFromInExpression(cond, allowedIdentifiers, dialect)
+
+	case runtimev1.Operation_OPERATION_AND:
+		return buildFromAndOrExpressions(cond, allowedIdentifiers, dialect, " AND ")
+
+	case runtimev1.Operation_OPERATION_OR:
+		return buildFromAndOrExpressions(cond, allowedIdentifiers, dialect, " OR ")
+
+	default:
+		leftExpr, args, err := buildFromExpression(cond.Exprs[0], allowedIdentifiers, dialect)
+		if err != nil {
+			return "", nil, err
+		}
+
+		rightExpr, subArgs, err := buildFromExpression(cond.Exprs[1], allowedIdentifiers, dialect)
+		if err != nil {
+			return "", nil, err
+		}
+		args = append(args, subArgs...)
+
+		return fmt.Sprintf("(%s) %s (%s)", leftExpr, conditionExpressionOperation(cond.Op), rightExpr), args, nil
+	}
+}
+
+func buildFromLikeExpression(cond *runtimev1.Condition, allowedIdentifiers map[string]identifier, dialect drivers.Dialect) (string, []any, error) {
+	if len(cond.Exprs) != 2 {
+		return "", nil, fmt.Errorf("like/not like expression should have exactly 2 sub expressions")
+	}
+
+	leftExpr, args, err := buildFromExpression(cond.Exprs[0], allowedIdentifiers, dialect)
+	if err != nil {
+		return "", nil, err
+	}
+
+	rightExpr, subArgs, err := buildFromExpression(cond.Exprs[1], allowedIdentifiers, dialect)
+	if err != nil {
+		return "", nil, err
+	}
+	args = append(args, subArgs...)
+
+	notKeyword := ""
+	if cond.Op == runtimev1.Operation_OPERATION_NLIKE {
+		notKeyword = "NOT"
+	}
+
+	// identify if immediate identifier has unnest
+	// TODO: do we need to do a deeper check?
+	unnest := false
+	ident, isIndent := cond.Exprs[0].Expression.(*runtimev1.Expression_Ident)
+	if isIndent {
+		i := allowedIdentifiers[ident.Ident]
+		unnest = i.unnest
+	}
+
+	var clause string
+	// Build [NOT] len(list_filter("dim", x -> x ILIKE ?)) > 0
+	if unnest && dialect != drivers.DialectDruid {
+		clause = fmt.Sprintf("%s len(list_filter(%s, x -> x %s ILIKE %s)) > 0", notKeyword, leftExpr, notKeyword, rightExpr)
+	} else {
+		if dialect == drivers.DialectDruid {
+			// Druid does not support ILIKE
+			clause = fmt.Sprintf("LOWER(%s) %s LIKE LOWER(%s)", leftExpr, notKeyword, rightExpr)
+		} else {
+			clause = fmt.Sprintf("%s %s ILIKE %s", leftExpr, notKeyword, rightExpr)
+		}
+	}
+
+	// When you have "dim NOT ILIKE '...'", then NULL values are always excluded.
+	// We need to explicitly include it.
+	if cond.Op == runtimev1.Operation_OPERATION_NLIKE {
+		clause += fmt.Sprintf(" OR %s IS NULL", leftExpr)
+	}
+
+	// TODO: is `col is null` needed?
```

Comment:
If it was in the previous implementation, it probably still is. Maybe `git blame` can help trace why it was added? (Good example of why it's nice to add comments for non-obvious code)

Reply:
It was needed. I added the code but didnt remove the TODO. Explanation in the comments above the if clause above this.